### PR TITLE
Add VM checks in OEM role

### PR DESCRIPTION
**Pull request by kylelaker**
As discussed in #189, there are some things that should not be done in
when the `oem` role is run on a physical machine. This adds a check for
being a virtualization guest for certain tasks.

### CODE REVIEW

File: roles/oem/tasks/main.yml [hunk 1]
```diff
@@ -66,3 +67,4 @@
   register: lspci_output
   failed_when: lspci_output.rc == 0
   changed_when: False
+  when: "ansbile_virtualization_role == 'guest'"
```
Comment (ripleymj) [inline roles/oem/tasks/main.yml]: Plz to "ansible"

**Comment by showalrm** [comment]
Does this not close #225? It seems like, looking at the list in that issue's discussion, that everything significant has been taken care of in some way.

**Comment by kylelaker** [comment]
@showalrm That's a good point. If @ripleymj agrees this does everything he wanted in that issue, I'll edit the initial commit.

**Comment by kylelaker** [comment]
I can't request changes on this PR since I opened it; however, the `dd` runs before the stuff to set up the config tool script. That probably wouldn't affect the goal of the `dd` too much.

If we don't care about this, go ahead and merge it.

If we do care, then we could probably either:

 - reverse the order of the roles in `oem.yml`
 - see if we can make the `dd` task a handler so it only ever runs once at the very end
    - can handlers have a `when`?
 - create a new role just for the `dd` and add it to the bottom of `oem.yml`.

**Comment by kylelaker** [comment]
My preference in the above is probably to reverse the order in `oem.yml`. I think it's okay if `common` stuff is run on not-Mint, so we don't necessarily need to check that before installing the config tool script and Tux.

**Comment by ripleymj** [comment]
Agreed, I think swapping them is better. Leaving it as is wouldn't hurt anything, as `common` does not remove anything to be zeroed. Moving `common` to be first in `oem.yml` makes this consistent with `local.yml`

**Comment by ripleymj** [comment]
Adding @showalrm as an unbiased reviewer.

**Comment by kylelaker** [comment]
I can't officially review, but FWIW, I approve.

**Comment by kylelaker** [comment]
@ripleymj Hopefully 171ef24 takes care of that or at least gets us close.

File: roles/oem/tasks/main.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@
   changed_when: False
   args:
     warn: no
+  when: "ansible_virtualization_role == 'guest'"
 - name: Ensure skeleton Desktop directory exists
```
Comment (ripleymj) [inline roles/oem/tasks/main.yml]: Do we need to keep this with it added to `common` now?

**Pull request merged**